junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/TestFactoryTestDescriptor.java

```diff
@@ -109,18 +112,20 @@ protected void invokeTestMethod(JupiterEngineExecutionContext context, DynamicTe
 	}
 
 	static Optional<JupiterTestDescriptor> createDynamicDescriptor(JupiterTestDescriptor parent, DynamicNode node,
-			int index, TestSource source, DynamicDescendantFilter dynamicDescendantFilter) {
+			int index, TestSource testSource, DynamicDescendantFilter dynamicDescendantFilter) {
 		UniqueId uniqueId;
 		Supplier<JupiterTestDescriptor> descriptorCreator;
+		TestSource source = node.getTestSource().orElse(testSource);
+		Set<TestTag> tags = node.getTags().stream().map(TestTag::create).collect(Collectors.toSet());
```

Comment:
We need some exception handling here because `TestTag::new` might throw an exception.